File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    
    let rlt = []
    for (let sc of score ) {
        if (rlt.length < k) { 
            rlt.push(sc)
            rlt = rlt.sort((a,b) => a-b)
            answer.push(rlt[0])
        }
        else {
            rlt = rlt.sort((a,b) => a-b)
            if (sc > rlt[0]) {
                rlt.shift()
                rlt.unshift(sc)
            }
            rlt = rlt.sort((a,b) => a-b)
            answer.push(rlt[0])
        }
    }
    return answer;
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    rank = []
    
    for s in score:
        rank.append(s)
        rank.sort(reverse = True)
        
        if len(rank) > k:
            # rank에서 k번째 수 밀어내기
            rank.pop()
            answer.append(min(rank))
        else:
            answer.append(min(rank))
    return answer